src/main/scala/ai/starlake/extract/ExtractSchema.scala

```diff
@@ -60,39 +59,82 @@ object ExtractSchema extends Extract with LazyLogging {
           case Failure(e) => throw e
         }
       }
-      extractSchema(jdbcSchema, connectionOptions, outputDir(config.outputDir), domainTemplate)
+      val currentDomain = schemaHandler.getDomain(jdbcSchema.schema, raw = true)
+      extractSchema(
+        jdbcSchema,
+        connectionOptions,
+        outputDir(config.outputDir),
+        domainTemplate,
+        currentDomain
+      )
     }
   }
 
   def extractSchema(
     jdbcSchema: JDBCSchema,
     connectionOptions: Map[String, String],
     baseOutputDir: File,
-    domainTemplate: Option[Domain]
+    domainTemplate: Option[Domain],
+    currentDomain: Option[Domain]
   )(implicit
     settings: Settings
   ): Unit = {
 
+    // End of list element have higher precedence
+    def mergeAllMetadata(metadatas: List[Metadata]): Metadata = {
+      metadatas.foldLeft(Metadata())(_.`import`(_))
+    }
+
     val domainName = jdbcSchema.schema.replaceAll("[^\\p{Alnum}]", "_")
     baseOutputDir.createDirectories()
     File(baseOutputDir, domainName).createDirectories()
-    val domain = extractDomain(jdbcSchema, connectionOptions, domainTemplate)
+    val extractedDomain = extractDomain(jdbcSchema, connectionOptions, domainTemplate)
+    val domain = extractedDomain.copy(metadata =
+      mergeAllMetadata(Nil ++ currentDomain.flatMap(_.metadata) ++ extractedDomain.metadata)
+        .copy(fillWithDefaultValue = false)
+        .asOption()
+    )
     val tables = domain.tables
     val tableRefs = tables.map("_" + _.name)
     tables.foreach { table =>
-      val tableWithWrite = jdbcSchema.write match {
-        case None => table
-        case Some(write) =>
-          val metadata =
-            domainTemplate.flatMap(_.metadata).getOrElse(Metadata()).copy(write = jdbcSchema.write)
-          table.copy(metadata = Some(metadata))
-      }
+      val restoredTable =
+        currentDomain.flatMap(_.tables.find(_.name == table.name)) match {
+          case Some(currentTable) =>
+            table.copy(
```

Comment:
I think that we should put this code in the Schéma class to centralize access to the fields there